api/users/serializers.py

```diff
@@ -128,6 +132,43 @@ def profile_image_url(self, user):
         size = self.context['request'].query_params.get('profile_image_size')
         return user.profile_image_url(size=size)
 
+    def _validate_user_json(self, value, json_schema):
+        try:
+            jsonschema.validate(value, from_json(json_schema))
+        except jsonschema.ValidationError as e:
+            if len(e.path) > 1:
+                raise InvalidModelValueError("For '{}' the field value {}".format(e.path[-1], e.message))
+
+            raise InvalidModelValueError(e.message)
+        except jsonschema.SchemaError as e:
+            raise InvalidModelValueError(e.message)
+
+        self._validate_dates(value)
+
+    def _validate_dates(self, info):
+        for history in info:
+
+            if history.get('startYear'):
+                startDate = datetime.date(history['startYear'], history['startMonth'], 1)
+
+            if not history['ongoing']:
+                if history.get('endYear'):
+                    endDate = datetime.date(history['endYear'], history.get('endMonth', 1), 1)
+
+                if history.get('startYear') and history.get('endYear'):
+                    if (endDate - startDate).days <= 0:
+                        raise InvalidModelValueError(detail='End date must be greater than or equal to the start date.')
+            elif history.get('endYear') or history.get('endMonth'):
+                raise InvalidModelValueError(detail='Ongoing positions cannot have end dates.')
```

Comment:
I'm hesitant to add this to the schema because:
A) It's hard for an outside observer to immediately know what this is for and I can't comment inside a JSON file to make this clearer.
B) The error message this produces isn't really human readable, getting back to the current message is likely going to include a lot of catching and re-throwing.

But there's definitely a benefit to always relying  using the schema, so 🤷‍♂️ .

Reply:
Yeah, I definitely agree that the error messages are way less than ideal. My opinion is that using the schema for validation is a lot nicer than writing our own, and we can document this in a more human-readable format in the dev docs.

Reply:
👍